### PR TITLE
add externalroom Venue template

### PR DESCRIPTION
**Pull request by ermiaSparkle**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/930

closes https://github.com/sparkletown/internal-sparkle-issues/issues/948

image

### CODE REVIEW

File: src/components/templates/ExternalRoom/ExternalRoom.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from "react";
+
+export interface ExternalRoomProps {
+  zoomUrl: string;
+}
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.tsx:5]:
```suggestion
export interface ExternalRoomProps {
  venue: WithId<AnyVenue>;
}

```

File: src/components/templates/ExternalRoom/ExternalRoom.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export const ExternalRoom: React.FC<ExternalRoomProps> = ({ zoomUrl }) => {
+  useEffect(() => {
+    window.open(zoomUrl);
+  }, [zoomUrl]);
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.tsx:9]: reuse utils functions to open external source

File: src/components/templates/ExternalRoom/ExternalRoom.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <div className="external-rooms">
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.tsx:11]: correct class naming as per standarts

File: src/pages/VenuePage/TemplateWrapper.tsx [hunk 4]
```diff
@@ -103,7 +104,25 @@ export const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
...
+            <button
+              role="link"
+              className="btn btn-primary"
+              onClick={() => history.goBack()}
+            >
+              Go Back
+            </button>
+          </p>
+        );
+      }
```
Comment (mike-lvov) [inline src/pages/VenuePage/TemplateWrapper.tsx:124]: this should be rendered inside the room

**Review by mike-lvov — CHANGES_REQUESTED** [review]

File: src/pages/VenuePage/TemplateWrapper.tsx (hunk 4 above)
Comment (ermiaSparkle) [reply src/pages/VenuePage/TemplateWrapper.tsx]: fixed in 03fcfc07bda6c3d8df4d15a25ea8f36bb1956118

File: src/components/templates/ExternalRoom/ExternalRoom.tsx (hunk 3 above)
Comment (ermiaSparkle) [reply src/components/templates/ExternalRoom/ExternalRoom.tsx]: fixed in 03fcfc07bda6c3d8df4d15a25ea8f36bb1956118

File: src/components/templates/ExternalRoom/ExternalRoom.tsx (hunk 2 above)
Comment (ermiaSparkle) [reply src/components/templates/ExternalRoom/ExternalRoom.tsx]: fixed in 03fcfc07bda6c3d8df4d15a25ea8f36bb1956118

File: src/components/templates/ExternalRoom/ExternalRoom.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export interface ExternalRoomProps {
+  venue: WithId<AnyVenue>;
+}
+
+export const ExternalRoom: React.FC<ExternalRoomProps> = ({ venue }) => {
+  const history = useHistory();
+
+  useEffect(() => {
+    openUrl(venue.zoomUrl ?? "");
+  }, [venue]);
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.tsx:19]:
```suggestion
const redirectUrl = venue.zoomUrl;

  useEffect(() => {
    if (!redirectUrl) return;
    
    openUrl(redirectUrl);
  }, [redirectUrl]);
```

File: src/components/templates/ExternalRoom/ExternalRoom.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      <p>
+        Venue {venue.name} should redirect to a URL, but none was set.
+        <br />
+        <button
+          role="link"
+          className="btn btn-primary"
+          onClick={() => history.goBack()}
+        >
+          Go Back
+        </button>
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.tsx:44]:
```suggestion
```
We already have a back button in the lop left corner

File: src/components/templates/ExternalRoom/ExternalRoom.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        </h4>
+        <h4>Return to this tab when you are ready to explore some more!</h4>
+        <a rel="noreferrer" href={venue.zoomUrl} target="_blank">
+          Open room manually
+        </a>
+      </div>
+    );
+  } else {
+    return (
+      <p>
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.tsx:35]: proper classname

File: src/pages/VenuePage/TemplateWrapper.tsx [hunk 8]
```diff
@@ -103,7 +104,9 @@ export const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
         );
       }
       break;
-
+    case VenueTemplate.externalroom:
```
Comment (mike-lvov) [inline src/pages/VenuePage/TemplateWrapper.tsx:107]: we don't need a new venueTemplate type. It should replace `zoomroom/artcar/etc`

File: src/pages/Admin/Venue/TemplateForm.tsx [hunk 9]
```diff
@@ -26,7 +26,7 @@ const templateImageMap: Record<VenueTemplate, string | undefined> = {
   [VenueTemplate.posterhall]: undefined,
   [VenueTemplate.posterpage]: undefined,
   [VenueTemplate.screeningroom]: undefined,
-
+  [VenueTemplate.externalroom]: undefined,
```
Comment (mike-lvov) [inline src/pages/Admin/Venue/TemplateForm.tsx:29]:
```suggestion
```

File: src/pages/Admin/Venue/TemplateForm.tsx [hunk 10]
```diff
@@ -53,6 +53,7 @@ const templateThumbImageMap: Record<VenueTemplate, string | undefined> = {
   [VenueTemplate.posterhall]: undefined,
   [VenueTemplate.posterpage]: undefined,
   [VenueTemplate.screeningroom]: undefined,
+  [VenueTemplate.externalroom]: undefined,
```
Comment (mike-lvov) [inline src/pages/Admin/Venue/TemplateForm.tsx:56]:
```suggestion
```

File: src/components/templates/ExternalRoom/ExternalRoom.scss [hunk 11]
```diff
@@ -0,0 +1,8 @@
+@import "scss/constants";
+
+.ExternalRoom__message {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin-top: $spacing--xxl;
+}
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.scss:8]:
```suggestion
.ExternalRoom {
&__message {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: $spacing--xxl;
}
}
```

**Review by mike-lvov — CHANGES_REQUESTED** [review]

File: src/pages/Admin/Venue/TemplateForm.tsx [hunk 12]
```diff
@@ -26,7 +26,6 @@ const templateImageMap: Record<VenueTemplate, string | undefined> = {
   [VenueTemplate.posterhall]: undefined,
   [VenueTemplate.posterpage]: undefined,
   [VenueTemplate.screeningroom]: undefined,
-
```
Comment (mike-lvov) [inline src/pages/Admin/Venue/TemplateForm.tsx:29]: why an empty line?

Comment (ermiaSparkle) [reply src/pages/Admin/Venue/TemplateForm.tsx:29]: I deleted the extra one was here

File: src/components/templates/ExternalRoom/ExternalRoom.tsx (hunk 7 above)
Comment (ermiaSparkle) [reply src/components/templates/ExternalRoom/ExternalRoom.tsx]: fixed in 733405113371f979b4941af7d375adc1f1e3678c

File: src/pages/VenuePage/TemplateWrapper.tsx (hunk 8 above)
Comment (ermiaSparkle) [reply src/pages/VenuePage/TemplateWrapper.tsx]: fixed in 733405113371f979b4941af7d375adc1f1e3678c

File: src/types/venues.ts [hunk 13]
```diff
@@ -35,7 +35,7 @@ export enum VenueTemplate {
   screeningroom = "screeningroom",
   themecamp = "themecamp",
   zoomroom = "zoomroom",
-
+  externalroom = "externalroom",
```
[the remark this answers is not in the document]
Comment (ermiaSparkle) [reply src/types/venues.ts]: fixed in 733405113371f979b4941af7d375adc1f1e3678c

File: src/components/atoms/Logo/Logo.tsx [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+}) => {
+  return (
+    <img
+      src={DEFAULT_VENUE_LOGO_PNG}
+      width={width}
+      height={height}
+      alt="Sparkle Logo"
+    />
+  );
+};
```
Comment (mike-lvov) [inline src/components/atoms/Logo/Logo.tsx:21]: We already have this component
image

File: src/components/templates/ExternalRoom/ExternalRoom.scss [hunk 15]
```diff
@@ -0,0 +1,29 @@
+@import "scss/constants";
+
+.ExternalRoom {
+  height: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  &__MessageBox {
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.scss:9]:
```suggestion
  &__message {
```

File: src/components/templates/ExternalRoom/ExternalRoom.scss [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+    align-items: center;
+    justify-content: space-between;
+    background: $light-grey;
+    padding: 70px;
+    padding-top: $spacing--xxl;
+    box-shadow: box-shadow--large(0.5);
+    border-radius: $border-radius--xl;
+  }
+
+  &__Header {
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.scss:22]:
```suggestion
  &__header {
```

File: src/components/templates/ExternalRoom/ExternalRoom.tsx [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+      </div>
+    );
+  } else {
+    return (
+      <div className="ExternalRoom__message">
+        <p>Venue {venue.name} should redirect to a URL, but none was set.</p>
+      </div>
+    );
+  }
+};
```
Comment (mike-lvov) [inline src/components/templates/ExternalRoom/ExternalRoom.tsx:43]:
```suggestion
    return redirectUrl ? (
      <div className="ExternalRoom">
        <div className="ExternalRoom__MessageBox">
          <Logo height={"110px"} width={"130px"} />
          <h4 className="ExternalRoom__Header">Thank your for visting</h4>
          <a rel="noreferrer" href={venue.zoomUrl} target="_blank">
            {venue.zoomUrl}
          </a>
        </div>
      </div>
    );
  : (
      <div className="ExternalRoom__message">
        <p>Venue {venue.name} should redirect to a URL, but none was set.</p>
      </div>
    );
  }
};
```

File: src/settings.ts [hunk 18]
```diff
@@ -50,6 +50,7 @@ export const SPARKLE_ICON = "/sparkle-icon.png";
 export const DEFAULT_MAP_BACKGROUND = "/maps/Sparkle_Field_Background.jpg";
 export const DEFAULT_VENUE_BANNER = "/assets/Sparkle_Banner_Default.jpg";
 export const DEFAULT_VENUE_LOGO = "/assets/Sparkle_SquareLogo_Default.jpg";
+export const DEFAULT_VENUE_LOGO_PNG = "/assets/Sparkle_SquareLogo_Default.png";
```
Comment (mike-lvov) [inline src/settings.ts:53]: Why don't you reuse the jpg one?

**Review by mike-lvov — CHANGES_REQUESTED** [review]

**Comment by mike-lvov** [comment]
@ermiaSparkle it should look like this
image

**Review by mike-lvov — APPROVED** [review]
LGTM

**Comment by whoaitsaimz** [comment]
Sorry but isn't the whole point of this to actually make the user go to the URL?  I don't see any opening content in a new tab happening here.

**Comment by mike-lvov** [comment]
> Sorry but isn't the whole point of this to actually make the user go to the URL? I don't see any opening content in a new tab happening here.

Happening where? Did you test this feature or just looked into the code?

**Comment by whoaitsaimz** [comment]
I tested-- but this one is on my end: Chrome disabled popups by default. I do not recall ever seeing a dialogue, perhaps it was a long time ago.  Setting "Allow" in my preferences did make the redirect work.  Thank you!

**Pull request merged**